src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -233,63 +230,62 @@ public String getDisplayName() {
+    private static AssumeRoleRequest.Builder createAssumeRoleRequest(String iamRoleArn, String iamExternalId) {
+        AssumeRoleRequest.Builder retval =
+                AssumeRoleRequest.builder().roleArn(iamRoleArn).roleSessionName("Jenkins");
         if (iamExternalId != null && !iamExternalId.isEmpty()) {
-            return retval.withExternalId(iamExternalId);
+            return retval.externalId(iamExternalId);
         }
         return retval;
     }
 
     /**
-     * Provides the {@link AWSSecurityTokenService} for a given {@link AWSCredentials}
+     * Provides the {@link StsClient} for a given {@link AwsCredentials}
      *
      * @param awsCredentials
-     * @return {@link AWSSecurityTokenService}
+     * @return {@link StsClient}
      */
-    private static AWSSecurityTokenService getAWSSecurityTokenService(AWSCredentials awsCredentials) {
-        ClientConfiguration clientConfiguration = getClientConfiguration();
-        String clientRegion = determineClientRegion();
-        return AWSSecurityTokenServiceClientBuilder.standard()
-                .withRegion(clientRegion)
-                .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
-                .withClientConfiguration(clientConfiguration)
+    private static StsClient getStsClient(AwsCredentials awsCredentials) {
+        SdkHttpClient clientConfiguration = getClientConfiguration();
+        Region clientRegion = determineClientRegion();
+        return StsClient.builder()
+                .region(clientRegion)
+                .credentialsProvider(StaticCredentialsProvider.create(awsCredentials))
+                .httpClient(clientConfiguration)
                 .build();
     }
 
     /**
-     * Provides the {@link ClientConfiguration}
+     * Provides the {@link SdkHttpClient}
      *
-     * @return {@link ClientConfiguration}
+     * @return {@link SdkHttpClient}
      */
-    private static ClientConfiguration getClientConfiguration() {
+    private static SdkHttpClient getClientConfiguration() {
         Jenkins instance = Jenkins.getInstanceOrNull();
 
-        ProxyConfiguration proxy = instance != null ? instance.proxy : null;
-        ClientConfiguration clientConfiguration = new ClientConfiguration();
+        hudson.ProxyConfiguration proxy = instance != null ? instance.proxy : null;
+        ApacheHttpClient.Builder builder = ApacheHttpClient.builder();
         if (proxy != null && proxy.name != null && !proxy.name.isEmpty()) {
-            clientConfiguration.setProxyHost(proxy.name);
-            clientConfiguration.setProxyPort(proxy.port);
-            clientConfiguration.setProxyUsername(proxy.getUserName());
-            clientConfiguration.setProxyPassword(proxy.getPassword());
+            builder.proxyConfiguration(ProxyConfiguration.builder()
+                    .endpoint(URI.create(String.format("%s:%s", proxy.name, proxy.port)))
```

Comment:
Not sure if this will work offhand. Do we need a scheme? Should be manually tested.

Reply:
Manually tested with a Squid reverse proxy.